### PR TITLE
Fix + add argument to pad for pytorch backend

### PR DESCRIPTION
**Pull request by FlorianPfaff**
## Checklist

- [X] My pull request has a clear and explanatory title.
- [X] If necessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [ ] I added appropriate unit tests.
- [ ] I made sure the code passes all unit tests. **No, but I'm not sure if that might be some random failure?**
- [X] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [X] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/main/docs/contributing.rst#coding-style-guidelines) and API.
- [ ] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/main/docs/contributing.rst#documentation))
- [ ] I linked to issues and PRs that are relevant to this PR.

## Description

In the pytroch backend, the argument used for padding values was "constant_value". However, the naming in jax and numpy is constant_value**s**. See https://numpy.org/doc/stable/reference/generated/numpy.pad.html
and https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.pad.html. I also added passing on the "mode" argument, which is supported by pytorch.

## Issue

This patch ensures that passing "constant_values" is compatible with all backends.

### CODE REVIEW

**Review by luisfpereira — APPROVED** [review]
Thank you @FlorianPfaff. Merging. (Sorry for taking so long.)
